sql/src/test/java/io/crate/execution/engine/join/HashInnerJoinBatchIteratorBehaviouralTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.join;
+
+import io.crate.breaker.RowAccounting;
+import io.crate.data.BatchIterator;
+import io.crate.data.Paging;
+import io.crate.data.Row;
+import io.crate.data.join.CombinedRow;
+import io.crate.testing.BatchSimulatingIterator;
+import io.crate.testing.TestingBatchIterators;
+import io.crate.testing.TestingRowConsumer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.mockito.Mockito.mock;
+
+public class HashInnerJoinBatchIteratorBehaviouralTest {
+
+    private int originalPageSize = Paging.PAGE_SIZE;
+
+    @Before
+    public void setUp() {
+        Paging.PAGE_SIZE = 2;
+    }
+
+    @After
+    public void tearDown() {
+        Paging.PAGE_SIZE = originalPageSize;
+    }
+
+    @Test
+    public void testHashJoinIteratorSwitchToRightWhenLeftExhausted() throws Exception {
+        BatchSimulatingIterator<Row> leftSimulatingIterator = new BatchSimulatingIterator<>(
+            TestingBatchIterators.ofValues(Arrays.asList(1, 2, 3, 4)), 2, 2, null);
```

Comment:
maxFakeBatches must be 1 (only 1 loadNextBatch call is valid for 4 rows with batchsize 2)